Source/WebCore/Modules/permissions/PermissionStatus.cpp

```diff
@@ -65,23 +65,15 @@ PermissionStatus::PermissionStatus(ScriptExecutionContext& context, PermissionSt
     : ActiveDOMObject(&context)
     , m_state(state)
     , m_descriptor(descriptor)
-    , m_source(source)
-    , m_page(WTFMove(page))
 {
     auto* origin = context.securityOrigin();
     auto originData = origin ? origin->data() : SecurityOriginData { };
-    m_origin = ClientOrigin { context.topOrigin().data(), WTFMove(originData) };
-    
-    if (source == PermissionQuerySource::Window) {
-        PermissionController::shared().addObserver(*this);
-        return;
-    }
+    auto clientOrigin = ClientOrigin { context.topOrigin().data(), WTFMove(originData) };
```

Comment:
`ClientOrigin clientOrigin { context.topOrigin().data(), WTFMove(originData) };`

Is more concise and looks nicer IMO.